comp/core/workloadmeta/collectors/util/kube_metadata_util_test.go

```diff
@@ -0,0 +1,89 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+// Package util contains utility functions for image metadata collection
+package util
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGenerateKubeMetadataEntityID(t *testing.T) {
+	tests := []struct {
+		name         string
+		namespace    string
+		resourceType string
+		resourceName string
+		expectedID   string
+	}{
+		{
+			name:         "namespace scoped resource",
+			namespace:    "default",
+			resourceType: "deployments",
+			resourceName: "app",
+			expectedID:   "deployments/default/app",
+		},
+		{
+			name:         "cluster scoped resource",
+			namespace:    "",
+			resourceType: "nodes",
+			resourceName: "foo-node",
+			expectedID:   "nodes//foo-node",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(tt *testing.T) {
+			assert.Equal(tt, GenerateKubeMetadataEntityID(test.resourceType, test.namespace, test.resourceName), test.expectedID)
```

Comment:
According to the `assert.Equal` docs, the second param is `expected` and the third `actual`.

Reply:
updated